src/main/java/eu/dissco/backend/repository/AnnotationRepository.java

```diff
@@ -21,80 +23,133 @@ public class AnnotationRepository {
     } catch (JsonProcessingException e) {
       log.error("Failed to parse annotation body to Json", e);
       return null;
     }
   }
 
+  private JsonApiData mapToJsonApiData(Record dbRecord) {
+    ObjectNode attributeNode = mapper.createObjectNode();
+    try {
+      attributeNode.put("id", dbRecord.get(NEW_ANNOTATION.ID));
+      attributeNode.put("version", dbRecord.get(NEW_ANNOTATION.VERSION));
+      attributeNode.put("type", dbRecord.get(NEW_ANNOTATION.TYPE));
+      attributeNode.put("motivation", dbRecord.get(NEW_ANNOTATION.MOTIVATION));
+      attributeNode.set("target", mapper.readTree(dbRecord.get(NEW_ANNOTATION.TARGET_BODY).data()));
+      attributeNode.set("body", mapper.readTree(dbRecord.get(NEW_ANNOTATION.BODY).data()));
+      attributeNode.put("preferenceScore", dbRecord.get(NEW_ANNOTATION.PREFERENCE_SCORE));
+      attributeNode.put("creator", dbRecord.get(NEW_ANNOTATION.CREATOR));
+      attributeNode.put("created", String.valueOf(dbRecord.get(NEW_ANNOTATION.CREATED)));
+      attributeNode.set("generator",
+          mapper.readTree(dbRecord.get(NEW_ANNOTATION.GENERATOR_BODY).data()));
+      attributeNode.put("generated", String.valueOf(dbRecord.get(NEW_ANNOTATION.GENERATED)));
+      attributeNode.put("deleted", String.valueOf(dbRecord.get(NEW_ANNOTATION.DELETED)));
+    } catch (JsonProcessingException e) {
+      log.error("Failed to parse annotation body to Json", e);
+      return null;
+    }
+    return new JsonApiData(dbRecord.get(NEW_ANNOTATION.ID), dbRecord.get(NEW_ANNOTATION.TYPE),
+        attributeNode);
+  }
 
   public List<AnnotationResponse> getForTarget(String id) {
-    return context.select(NEW_ANNOTATION.asterisk())
-        .distinctOn(NEW_ANNOTATION.ID)
-        .from(NEW_ANNOTATION)
-        .where(NEW_ANNOTATION.TARGET_ID.eq(id))
+    return context.select(NEW_ANNOTATION.asterisk()).distinctOn(NEW_ANNOTATION.ID)
+        .from(NEW_ANNOTATION).where(NEW_ANNOTATION.TARGET_ID.eq(id))
         .and(NEW_ANNOTATION.DELETED.isNull())
-        .orderBy(NEW_ANNOTATION.ID, NEW_ANNOTATION.VERSION.desc())
-        .fetch(this::mapToAnnotation);
+        .orderBy(NEW_ANNOTATION.ID, NEW_ANNOTATION.VERSION.desc()).fetch(this::mapToAnnotation);
   }
 
   public List<Integer> getAnnotationVersions(String id) {
     return context.select(NEW_ANNOTATION.VERSION).from(NEW_ANNOTATION)
-        .where(NEW_ANNOTATION.ID.eq(id)).fetch(
-            Record1::value1).stream().toList();
+        .where(NEW_ANNOTATION.ID.eq(id)).fetch(Record1::value1).stream().toList();
   }
 
   public int getAnnotationForUser(String id, String userId) {
-    return context.select(NEW_ANNOTATION.ID)
-        .distinctOn(NEW_ANNOTATION.ID)
-        .from(NEW_ANNOTATION)
-        .where(NEW_ANNOTATION.ID.eq(id))
-        .and(NEW_ANNOTATION.CREATOR.eq(userId))
+    return context.select(NEW_ANNOTATION.ID).distinctOn(NEW_ANNOTATION.ID).from(NEW_ANNOTATION)
```

Comment:
I would rather have a bit more overview here and have a newline after each statement